src/Kmd.Logic.Gateway.Automation/PublishApis.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Client;
+using Kmd.Logic.Gateway.Automation.Client.Models;
+using Kmd.Logic.Gateway.Automation.PublishFile;
+using Kmd.Logic.Identity.Authorization;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    internal class PublishApis
+    {
+        private readonly PublishPolicies _publishPolicies;
+        private readonly GatewayOptions options;
+        private readonly GatewayClientFactory gatewayClientFactory;
+        private List<GatewayAutomationResult> _publishResults;
+
+        public PublishApis(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, GatewayOptions options, List<GatewayAutomationResult> publishResults)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
```

Comment:
If these check we are already doing in calling method do we required again here ?

Reply:
I had to check to avoid warning.